pipeline/tasks/io_tasks.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tasks related to I/O."""
+
+import base64
+import os
+import subprocess
+import shutil
+import urllib
+import zipfile
+
+from gcloud import storage
+from oauth2client.client import GoogleCredentials
+from pipeline.tasks import task_base
+
+DEFAULT_PROJECT = 'vkit-pipeline'
+
+
+def get_credential():
+    return GoogleCredentials.get_application_default()
+
+
+class BlobUploadTask(task_base.TaskBase):
+    """A task which uploads file to Google Cloud Storage.
+
+    It requires authentication be properly configured."""
+
+    default_provides = ('bucket', 'path', 'public_url')
+
+    def execute(self,
+                bucket_name,
+                src_path,
+                dest_path,
+                project=DEFAULT_PROJECT):
+        print "Start blob upload"
+        client = storage.Client(project=project, credentials=get_credential())
+        bucket = client.get_bucket(bucket_name)
+        blob = bucket.blob(dest_path)
+        with open(src_path, 'r') as f:
+            blob.upload_from_file(f)
+        print "Uploaded to %s" % blob.public_url
+
+        return bucket_name, dest_path, blob.public_url
+
+
+class BlobDownloadTask(task_base.TaskBase):
+    """A task which downloads file to Google Cloud Storage.
+
+  It requires authentication be properly configured."""
+
+    def execute(self, bucket_name, path, output_dir, project=DEFAULT_PROJECT):
+        client = storage.Client(project=project, credentials=get_credential())
+        bucket = client.get_bucket(bucket_name)
+        blob = bucket.get_blob(path)
+        if blob:
```

Comment:
I personally like the style of:
```
if not blob:
    print 'Cannot find the output from GCS.'
    return
filename = ...
...
```
This reduces the indent.

Also -- is it good to accept nonexisting blobs?  Should we raise an error?